src/settings/settings.go

```diff
@@ -49,6 +49,14 @@ type Settings struct {
 	CacheKeyPrefix             string  `envconfig:"CACHE_KEY_PREFIX" default:""`
 	BackendType                string  `envconfig:"BACKEND_TYPE" default:"redis"`
 
+	// Settings for optional returning of custom headers
+	// value: the current limit
+	HeaderRatelimitLimit string `envconfig:"LIMIT_LIMIT_HEADER" default:""`
+	// value: remaining count
+	HeaderRatelimitRemaining string `envconfig:"LIMIT_REMAINING_HEADER" default:""`
+	// value: remaining seconds
+	HeaderRatelimitReset string `envconfig:"LIMIT_RESET_HEADER" default:""`
```

Comment:
IMO I think it would be better to have an additional flag, something like `XRatelimitResponseHeadersEnabled` rather than having the toggle logic in the service struct and have these 3 settings have a default value corresponding to the header names specified in the RFC, eg: "X-RateLimit-Limit" "X-RateLimit-Remaining" and "X-RateLimit-Reset"

Reply:
Sure, I can go with that

Reply:
I guess what I'm trying to say is -- most likely 99% of people will use the official header names as stated in the RFC/Envoy docs so I think we should set those as the default values in here. That means we would need an 'enabled' flag in the settings to determine if these responses are enabled rather than enabling them if the strings are not empty. I think that would be cleaner as well.

Reply:
Yes, I agree I'll do that change with an "Enable" flag, but the default values to be identical to the ones that are generated by default by Envoy might be a bit unecessary. In which case enabling the custom headers, will generate the same headers as Envoy does by default. I can do that if you think it makes sense, but perhaps it would make more sense to use the proper-headers from the draft spec, ie. without the "X-" (https://datatracker.ietf.org/doc/html/draft-polli-ratelimit-headers-03)

Reply:
Added the draft standard headers as the default values